eth/eventservices/rewardservice.go

```diff
@@ -99,44 +106,26 @@ func (s *RewardService) tryReward() error {
 	}
 
 	if t.LastRewardRound.Cmp(currentRound) == -1 && initialized && active {
-		var (
-			tx  *types.Transaction
-			err error
-		)
-
-		if s.pendingTx != nil {
-			// Previous attempt to call reward() still pending
-			// Replace pending tx by bumping gas price
-			tx, err = s.client.ReplaceTransaction(s.pendingTx, "reward", nil)
-			if err != nil {
-				if err == eth.ErrReplacingMinedTx || err.Error() == "nonce too low" {
-					// Pending tx confirmed so we should not try to replace next time
-					s.pendingTx = nil
-				}
-
-				return err
-			}
-		} else {
-			// No previous attempt to call reward(), invoke with next nonce
-			tx, err = s.client.Reward()
-			if err != nil {
-				return err
-			}
+		tx, err := s.client.Reward()
+		if err != nil {
+			return err
 		}
 
-		s.pendingTx = tx
-
 		err = s.client.CheckTx(tx)
 		if err != nil {
 			if err == context.DeadlineExceeded {
-				glog.Infof("Reward tx did not confirm within defined time window - will try to replace pending tx next time")
+				glog.Infof("Reward tx did not confirm within defined time window - will try to replace pending tx once")
+				// Previous attempt to call reward() still pending
+				// Replace pending tx by bumping gas price
+				tx, err = s.client.ReplaceTransaction(tx, "reward", nil)
+				if err != nil {
+					return err
+				}
```

Comment:
Calling `checkTx()` for a replacement tx and returning an error if it fails in 2a5a59a099c35208140ab7083ab177af82aa29d9 

However we currently call `tryReward()` when receiving a new round event instead of a new block , ideally the next call will be to call reward for the round after any still pending tx. If replacing the tx fails the user will have to speed it up manually if necessary. We can implement retries when we implement a `MaxGasPrice` #1671

Reply:
So, the reason for only replacing a reward tx once and not supporting multiple replacements for the scenario where the first replacement does not confirm within a specific time window is that `tryReward()` is now only called when a NewRound event is emitted instead of at a regular polling interval. Got it. I think that's a reasonable change in behavior for now especially because replacement transactions without a gas price cap historically haven't been particularly helpful for users (if anything they've caused more harm in some cases).

Reply:
Yes, pretty much. We could loop but without a gas cap costs could run up rapidly under rare circumstances. I feel like the gas price oracle does a fairly good job of recommending a gas price that under recent conditions should confirm within a couple of blocks.